### PR TITLE
Use new boundary input

### PR DESCRIPTION
**Pull request by RevathiJambunathan**
This PR adds an abort if old boundary condition is used.
The input files used for all CI-tests and those in the examples in the repo are updated with the new boundary interface.

### CODE REVIEW

File: Examples/Tests/SingleParticle/inputs_2d [hunk 1]
```diff
@@ -4,14 +4,18 @@ amr.max_level = 0
...
+
+# Boundary condition
+boundary.field_lo = pec pec
+boundary.field_hi = pec pec
+
 algo.charge_deposition = standard
 algo.field_gathering = energy-conserving
 warpx.cfl = 1.0
 
+
```
Comment (RevathiJambunathan) [inline Examples/Tests/SingleParticle/inputs_2d]:
```suggestion
```

**Review by dpgrote — APPROVED** [review]
Looks good! Thanks for all of the work on this!

File: Examples/Physics_applications/laser_acceleration/inputs_2d_rz [hunk 2]
```diff
@@ -6,21 +6,22 @@ amr.n_cell =  64  512
...
 geometry.coord_sys   = 1                  # 0: Cartesian
-geometry.is_periodic = 0     0      # Is periodic?
 geometry.prob_lo     =   0.   -56.e-6    # physical domain
 geometry.prob_hi     =  30.e-6    12.e-6
 amr.max_level = 0 # Maximum level in hierarchy (1 might be unstable, >1 is not supported)
 
 warpx.n_rz_azimuthal_modes = 2
 
+boundary.field_lo = none pec
+boundary.field_hi = none pec
```
Comment (RevathiJambunathan) [inline Examples/Physics_applications/laser_acceleration/inputs_2d_rz]: @dpgrote for this RZ input, is none the right boundary condition for R-max?

Comment (RevathiJambunathan) [reply Examples/Physics_applications/laser_acceleration/inputs_2d_rz]:
```suggestion
boundary.field_lo = none pec
boundary.field_hi = pec pec
```

**Comment by ax3l** [comment]
offline discussed: `PEC_particle` reset because the old checked in version had PMLs accidentally on alongside.

**Review by ax3l — APPROVED** [review]
Thanks a lot, fantastic to have all the boundaries together in one place :rocket: :sparkles:

**Comment by RevathiJambunathan** [comment]
@ax3l thank you for the discussion offline! I have added a comment e981a4d to clarify how periodicity is set and that it is required by amrex for defining geometry and for communication.
The default parameters for field and particle boundary are now declared in the very first line in the doc. 2dd3389

**Review by ax3l — APPROVED** [review]
Thank you for the offline clarification and subsequent documentation update :+1: :tada:
